### ISSUE TITLE
Panic when null value used as default in var with type `any`

### ISSUE THREAD

**Issue by robison** [hashicorp/terraform#22334]
### Terraform Version
```
Terraform v0.12.6
```

### Terraform Configuration Files

```hcl
terraform {
  required_version = ">= 0.12.6"
}

variable "list" {
  type = list(object({
    name    = string
    records = list(string)
    ttl     = any
    type    = string
  }))
  default = [
    {
      name    = "aaaa"
      records = ["224.0.0.0"]
      ttl     = 900
      type    = "A"
    },
    {
      name    = "bbbb"
      records = ["239.0.0.0"]
      ttl     = 3600
      type    = "CNAME"
    },
    {
      name    = "cccc"
      records = ["240.0.0.0"]
      ttl     = null
      type    = "CNAME"
    },
  ]
  description = "List of DNS record objects to manage."
}

output "list" {
  value = var.list
}
```

### Crash Output

https://gist.github.com/robison/e287324916c2db0f3d482459c2eb5db1

### Expected Behavior
Output:
```
Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

list = [
  {
    "name" = "aaaa"
    "records" = [
      "224.0.0.0",
    ]
    "ttl" = 900
    "type" = "A"
  },
  {
    "name" = "bbbb"
    "records" = [
      "239.0.0.0",
    ]
    "ttl" = 3600
    "type" = "CNAME"
  },
  {
    "name" = "cccc"
    "records" = [
      "240.0.0.0",
    ]
    "type" = "CNAME"
  },
]
```
Note: this is the output from when the attribute type for `ttl = number` - I somewhat didn't expect the attribute to be erased, but rather some other behavior, whether set to `0` or some other output.
### Actual Behavior
Terraform panicked.

### Steps to Reproduce

`terraform init` and `terraform apply` will both panic with this condition set.

**Comment by danieldreier** [comment]
@robison I want to apologize for how long it took to respond to this crash report. I really appreciate you reporting this. I just went and was able to reproduce this in terraform 0.12.26. Fortunately, I was unable to reproduce it in 0.13.0 beta 1, which indicates it's fixed in 0.13.0. I'm going to resolve this because it seems to be fixed. Please let me know (ddreier@hashicorp.com) if you're still seeing this in 0.13.0 beta 1 and I'm happy to re-open and reinvestigate this. Thank you again for reporting this!

**Comment by robison** [comment]
@danieldreier Is 0.12.26 the last release in 0.12?
